plugins/CoreUpdater/Updater.php

```diff
@@ -196,6 +200,36 @@ public function oneClickUpdatePartTwo($newVersion = null)
             }
         }
 
+        // get a list of previously activated plugins and try to reactivate them if there are no missing requirements
+        try {
+            $previouslyActivePlugins = Option::get('OneClickUpdate_ActivatedPlugins');
+            if (false !== $previouslyActivePlugins) {
+                $previouslyActivePlugins = json_decode($previouslyActivePlugins, true);
+            } else {
+                $previouslyActivePlugins = [];
+            }
+
+            $reactivatedPlugins = [];
+            if (!empty($previouslyActivePlugins)) {
+                $pluginManager = PluginManager::getInstance();
+                foreach ($previouslyActivePlugins as $previouslyActivePluginName) {
+                    if (!$pluginManager->isPluginActivated($previouslyActivePluginName)) {
+                        $plugin = $pluginManager->loadPlugin($previouslyActivePluginName);
+                        if (empty($plugin->getMissingDependencies($newVersion))) {
+                            $plugin->activate();
```

Comment:
```suggestion
                            $pluginManager->activatePlugin($plugin);
```

When I tried the updated process locally, the plugin was updated, announced as "reactivated", but missing in the `[Plugins]` config. It was kept in `[PluginsInstalled]`.

Just having the plugin manager take care of updating the config should prevent that, unless it should have never been removed in the first place and something else was wrong (and is not reproducible).

Reply:
Thanks for finding that quirk, I've updated it to use plugins manager as suggested, with a tweak that it takes the plugin name as a param, not the param object.